packages/packages_linux.go

```diff
@@ -177,3 +179,20 @@ func getInstalledPackages(ctx context.Context) (Packages, []string) {
 
 	return pkgs, errs
 }
+
+// NewInstalledPackagesProvider makes provider that uses osv-scalibr as its implementation if enabled by config, otherwise falls back to default legacy implementation.
+func NewInstalledPackagesProvider(osinfoProvider osinfo.Provider) InstalledPackagesProvider {
```

Comment:
There small problem that current scenario does not handle, my assumption that inventory extraction logic should not be changed after the scalibr enablement, but in this scenario if SCALIBR enabled i assume we will trigger only 3 currently supported SCALIBR extractors, so we will broke current logic.